### PR TITLE
GET Events

### PR DESCRIPTION
**Pull request by Andreas-Garcia**
- GET Events

### CODE REVIEW

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+/**
+ * Dao for Events
+ * @author Andréas Garcia <andreas.garcia@inra.fr>
+ */
+public class EventDAOSesame extends DAOSesame<Event> {
+    final static Logger LOGGER = LoggerFactory.getLogger(EventDAOSesame.class);
+    
+    // constants used for SPARQL variables
+    private static final String URI_VARIABLE = "uri";
+    private static final String URI_VARIABLE_SPARQLE = "?" + URI_VARIABLE;
```
[the remark this answers is not in the document]
Comment (Andreas-Garcia) [reply phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: what is the issue more specifically ?

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/PropertyDAOSesame.java [hunk 2]
```diff
@@ -643,16 +685,38 @@ protected SPARQLQueryBuilder prepareSearchPropertiesQuery(String language) {
...
+     * and fill the RDF Resource defintion object with the values and labels
+     * 
+     * @param definition The definition object which will be filled
+     * @param language specify in which language labels should be returned.
+     * @param propertiesRelationsToIgnore
      * @return true    if the definition object is correctly filled
      *          false   if the uri doesn't exists
      */
-    public boolean getAllPropertiesWithLabels(RdfResourceDefinition definition, String language) {
+    public boolean getRdfObjectPropertiesExceptThoseSpecifiedAndAddThemToIt(
```
[the remark this answers is not in the document]
Comment (Andreas-Garcia) [reply phis2-ws/src/main/java/phis2ws/service/dao/sesame/PropertyDAOSesame.java]: - Modification of the purpose : the old method counldn't ignore relations we don't want in the result (such has "concerns" in the case of the events, we don't consider it as a property). I just add this possibility to ignore relations. The object that previously called this method just don't use this new possibility so their  functionning isn't altered.
- Modification of the name : 
   - I specify that the function set the properties to the object in parameter
   -  I add the new feature in the name too to tell what it does

[the remark this answers is not in the document]
Comment (Andreas-Garcia) [reply phis2-ws/src/main/java/phis2ws/service/dao/sesame/PropertyDAOSesame.java]: I set the old name in the new pull request

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/RadiometricTargetDAOSesame.java [hunk 3]
```diff
@@ -89,7 +89,8 @@ protected SPARQLQueryBuilder prepareSearchQuery() {
      * Get a radiometric target from a given binding set.
```
[the remark this answers is not in the document]
Comment (Andreas-Garcia) [reply phis2-ws/src/main/java/phis2ws/service/dao/sesame/RadiometricTargetDAOSesame.java]: a refactoring of the getAllPropertiesWithLabels made modifications to this file

File: phis2-ws/src/main/java/phis2ws/service/dao/manager/DAOSesame.java [hunk 4]
```diff
@@ -62,14 +68,28 @@
...
+    protected static final String DATETIME_VARIABLE = "dateTime";
+    protected static final String DATETIME_VARIABLE_SPARQL = "?" + DATETIME_VARIABLE;
+    
+    protected static final String DATE_RANGE_START_DATETIME_VARIABLE = "dateRangeStartDateTime";
+    protected static final String DATE_RANGE_START_DATETIME_VARIABLE_SPARQL = "?" + DATE_RANGE_START_DATETIME_VARIABLE;
+    
+    protected static final String DATE_RANGE_END_DATETIME_VARIABLE = "dateRangeEndDateTime";
+    protected static final String DATE_RANGE_END_DATETIME_VARIABLE_SPARQL = "?" + DATE_RANGE_END_DATETIME_VARIABLE;
+    
+    protected final String DATETIMESTAMP_FORMAT_SPARQLE = DateFormat.YMDTHMSZZ.toString();
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/manager/DAOSesame.java]: DATETIMESTAMP_FORMAT_SPARQLE to DATETIMESTAMP_FORMAT_SPARQL

File: phis2-ws/src/main/java/phis2ws/service/dao/manager/DAOSesame.java [hunk 5]
```diff
@@ -277,6 +297,81 @@ public String getValueFromPredicate(String subject, String predicate) throws Rep
...
+     * @param filterDateString
+     * @param filterDateFormat
+     * @param filterDateSparqlVariable SPARQL variable (?abc format)
+     * @param comparisonSign e.g >, >=, <, <= 
+     * @param dateTimeStampToCompareSparqlVariable the SPARQL variable 
+     * (?abc format) of the dateTimeStamp to which the date has to be compared
+     * @example SparQL code added to the query :
+        BIND(xsd:dateTime(str("2017-09-10T12:00:00+01:00")) as ?dateRangeStartDateTime) .
+        FILTER ( (?dateRangeStartDateTime <= ?dateTime) ) 
+
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/manager/DAOSesame.java]: remove blank line

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+     * Set a search query to select a type and to filter according to it 
+     * if necessary
+     * @example SparQL filter added :
+     *  ?type  <http://www.w3.org/2000/01/rdf-schema#subClassOf>*  <http://www.phenome-fppn.fr/vocabulary/2018/oeev#MoveFrom> . 
+     */
+    private void prepareSearchQueryType(SPARQLQueryBuilder query, String sparqlVariableUri, String searchType){
+        query.appendSelect(TYPE_VARIABLE_SPARQL);
+        query.appendGroupBy(TYPE_VARIABLE_SPARQL);
+        query.appendTriplet(sparqlVariableUri
+            , Rdf.RELATION_TYPE.toString(), TYPE_VARIABLE_SPARQL, null);
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: could you change the indentation

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java [hunk 7]
```diff
@@ -0,0 +1,505 @@
...
+                    + ", \"" + searchConcernsItemUri + "\", \"i\")");
+            }
+        }
+    }
+
+    /**
+     * Set a search query to select a datetime from an instant and to filter 
+     * according to it if necessary
+     * @example SparQL filter added :
+     * ?uri  <http://www.w3.org/2006/time#hasTime>  ?time  .
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: please add select clause

Comment (annetireau) [reply phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: and group by

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java [hunk 8]
```diff
@@ -0,0 +1,505 @@
...
+     * BIND(<http://www.w3.org/2001/XMLSchema#dateTime>(str(?dateTimeStamp)) as ?dateTime) .
+     * BIND(<http://www.w3.org/2001/XMLSchema#dateTime>(str("2017-09-10T12:00:00+01:00")) as ?dateRangeStartDateTime) .
+     * BIND(<http://www.w3.org/2001/XMLSchema#dateTime>(str("2017-09-12T12:00:00+01:00")) as ?dateRangeEndDateTime) .
+     */
+    private void prepareSearchQueryDateTime(SPARQLQueryBuilder query, String sparqlVariableUri, String searchDateTimeRangeStartString, String searchDateTimeRangeEndString){  
+        
+        query.appendSelect(DATETIMESTAMP_VARIABLE_SPARQL);
+        query.appendGroupBy(DATETIMESTAMP_VARIABLE_SPARQL);
+        query.appendTriplet(
+                sparqlVariableUri
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: please write inline

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java [hunk 9]
```diff
@@ -0,0 +1,505 @@
...
+     *  LIMIT 20 
+     *  OFFSET 0 
+     */
+    protected SPARQLQueryBuilder prepareSearchQuery(Event eventSearchParameters, String searchConcernsItemLabel, String searchConcernsItemUri, String dateRangeStartString, String dateRangeEndString, User user) {
+        SPARQLQueryBuilder query = new SPARQLQueryBuilder();
+        query.appendDistinct(Boolean.TRUE);
+        
+        String sparqlVariableUri = prepareSearchQueryUri(query, eventSearchParameters.getUri());
+        prepareSearchQueryType(query, sparqlVariableUri, eventSearchParameters.getType());  
+        prepareSearchQueryConcernsItemFilter(query, sparqlVariableUri,
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: please inline

File: phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java [hunk 10]
```diff
@@ -0,0 +1,505 @@
...
+     * @param dateRangeStartString
+     * @param dateRangeEndString
+     * @param user
+     * @param searchPage
+     * @param searchPageSize
+     * @return events
+     */
+    public ArrayList<Event> searchEvents(Event eventSearchParameters, String searchConcernsItemLabel, String searchConcernsItemUri, String dateRangeStartString, String dateRangeEndString, User user, int searchPage, int searchPageSize) {
+        
+        SPARQLQueryBuilder eventsQuery = prepareSearchQuery(
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/sesame/EventDAOSesame.java]: inline please

**Pull request merged**